donation_bank_statement/models/bank_statement.py

```diff
@@ -0,0 +1,124 @@
+            }
+        vals = {
+            'company_id': company.id,
+            'partner_id': stline.partner_id.id,
+            'journal_id': company.donation_credit_transfer_journal_id.id,
+            'currency_id': company.currency_id.id,
+            'payment_ref': _('Credit Transfer %s') % stline.ref,
+            'check_total': amount,
+            'donation_date': stline.date,
+            'campaign_id': False,
+            'bank_statement_line_id': stline.id,
+            'line_ids': [(0, 0, line_vals)],
+            }
+        return vals
+
+    def create_donations(self):
+        self.ensure_one()
+        ddo = self.env['donation.donation']
+        prec = self.currency_id.rounding
+        if not self.company_id.donation_credit_transfer_journal_id:
+            # Don't raise an error here, because we may be in a multi-company
+            # setup where only some companies handle donations, not all
+            logger.info(
+                "No journal for donations via credit transfer for the "
+                "company %s: not creating donations" % self.company_id.name)
+            return False
+        journal = self.company_id.donation_credit_transfer_journal_id
+        if not journal.default_debit_account_id:
+            raise UserError(_(
+                "Missing Default Debit Account on the Journal '%s'.")
+                % journal.name)
+        donation_ids = []
+        transit_account = journal.default_debit_account_id
+        for stline in self.line_ids:
+            if (
+                    float_compare(
+                        stline.amount, 0, precision_rounding=prec) == 1 and
+                    not stline.donation_ids):
+                for amline in stline.journal_entry_ids:
+                    for mline in amline:
+                        if (
+                                float_compare(
+                                    mline.credit, 0,
+                                    precision_rounding=prec) == 1 and
+                                mline.account_id == transit_account and
+                                not mline.reconciled):
+                            if not stline.partner_id:
+                                raise UserError(_(
+                                    "Missing partner on bank statement line "
+                                    "'%s' dated %s with amount %s.")
+                                    % (stline.name, stline.date,
+                                        stline.amount))
+                            vals = self._prepare_donation_vals(stline, mline)
+                            donation = ddo.create(vals)
+                            donation.partner_id_change()
+                            donation.line_ids.product_id_change()
+                            donation_ids.append(donation.id)
+
+    @api.multi
+    def button_confirm_bank(self):
+        statements = self.filtered(lambda st: st.state == 'open')
+        for stmt in statements:
```

Comment:
```suggestion
        for stmt in self.filtered(lambda st: st.state == 'open'):
```

There is no need for the variable

Reply:
Thanks, Done